packages/edit-site/src/components/list/added-by.js

```diff
@@ -20,6 +20,94 @@ import { _x } from '@wordpress/i18n';
 
 const TEMPLATE_POST_TYPE_NAMES = [ 'wp_template', 'wp_template_part' ];
 
+/**
+ *
+ * @typedef AddedByData
+ * @type {Object}
+ * @property {JSX.Element} icon         The icon to display.
+ * @property {string}      [imageUrl]   The optional image URL to display.
+ * @property {string}      [text]       The text to display.
+ * @property {boolean}     isCustomized Whether the template has been customized.
+ *
+ * @param    {Object}      template     The template object.
+ * @return {AddedByData} The added by object or null.
+ */
+export function useAddedBy( template ) {
+	return useSelect(
+		( select ) => {
+			const { getTheme, getPlugin, getEntityRecord, getMedia, getUser } =
+				select( coreStore );
+			if ( TEMPLATE_POST_TYPE_NAMES.includes( template.type ) ) {
+				// Added by theme.
+				// Template originally provided by a theme, but customized by a user.
+				// Templates originally didn't have the 'origin' field so identify
+				// older customized templates by checking for no origin and a 'theme'
+				// or 'custom' source.
+				if (
+					template.has_theme_file &&
+					( template.origin === 'theme' ||
+						( ! template.origin &&
+							[ 'theme', 'custom' ].includes(
+								template.source
+							) ) )
+				) {
+					return {
+						icon: themeIcon,
+						text:
+							getTheme( template.theme )?.name?.rendered ||
+							template.theme,
+						isCustomized: template.source === 'custom',
+					};
+				}
+
+				// Added by plugin.
+				// Template originally provided by a plugin, but customized by a user.
```

Comment:
I think the part about 'customized by a user' is incorrect (as it was in `trunk`), as we can see there's an `isCustomized` property lower down to determine this. It might be a good opportunity to remove it.

```suggestion
```